pkg/sql/sem/tree/eval.go

```diff
@@ -5228,6 +5228,23 @@ func SimilarToEscape(ctx *EvalContext, unescaped, pattern, escape string) (Datum
 	return matchRegexpWithKey(ctx, NewDString(unescaped), key)
 }
 
+// SimilarPattern converts a SQL regexp 'pattern' to a POSIX regexp 'pattern' using custom escape token 'escape'
+// which must be either empty (which disables the escape mechanism) or a single unicode character.
+func SimilarPattern(pattern, escape string) (Datum, error) {
+	var escapeRune rune
```

Comment:
this code is very similar to `SimilarToEscape`. can you abstract the point up to `similarToKey{s: pattern, escape: escapeRune}` into a separate function?

Reply:
OK, I have abstracted a function called `makeSimilarToKey`.